### PR TITLE
Document Object Model

### PR DESCRIPTION
**Pull request by SableVector**
### Document Object Model

[demo](https://sablevector.github.io/myPortfolio/kottansFrontend/document_object_model/)

[Code base](https://github.com/SableVector/myPortfolio/tree/main/kottansFrontend/document_object_model)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by SableVector** [comment]
I’ve completed the self-assessment and corrected the code accordingly. :)

File: submissions/SableVector/main.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+        let listItems = '';
+
+        arr.forEach(item => {
+            const { title } = item;
+
+            listItems += `<li class="sidebar-list__item">
+                            <button class="sidebar-list-item__button">${title}</button>
+                        </li>`;
+        });
```
Comment (madmaxWMFU) [inline submissions/SableVector/main.js:142]: Try to use the method `reduce ` of the _Array_

File: submissions/SableVector/main.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+        if (target && target.closest('.sidebar-list-item__button')) {
+            e.preventDefault();
+
+            // delete content
+            removeBook();
+
+            // create content
+            books.find(book => book.title === target.textContent ? createBook(book) : null);
+        }
```
Comment (madmaxWMFU) [inline submissions/SableVector/main.js:123]:
```suggestion
            const selectedBook = books.find(({title}) => title === target.textContent);
            createBook(selectedBook);
        }
```

File: submissions/SableVector/main.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+        createBook(books[0]);
+    } else {
+        createBook(books[0]);
+    }
+
+    // create content on the page after clicking on the button
+    sidebar.addEventListener('click', function (e) {
+        const target = e.target;
+
+        if (target && target.closest('.sidebar-list-item__button')) {
```
Comment (madmaxWMFU) [inline submissions/SableVector/main.js:115]:
```suggestion
        if (target.closest('.sidebar-list-item__button')) {
```

File: submissions/SableVector/main.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    // create buttons on the page
+    createButton(books);
+
+    // create first book on the page
+    if (pageContainer.querySelector('.book')) {
+        removeBook();
+        createBook(books[0]);
+    } else {
+        createBook(books[0]);
+    }
```
Comment (madmaxWMFU) [inline submissions/SableVector/main.js:109]:
```suggestion
    const [initialBook] = books;
    if (pageContainer.querySelector('.book')) {
        removeBook();
    } 
    createBook(initialBook);
```

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work! See some comments below.

**Pull request closed without merging**